content/competitionPlatform/howToSubmitAnAgent/submitYourOwnAgent/_index.en.md

```diff
@@ -4,9 +4,92 @@ weight: 50
+
+1. Store your agent files (e.g. scripts and weights) in a private model
+2. Create your personal access token (<a href="https://huggingface.co/docs/hub/security-tokens" target="_blank">official docs here</a>):
+   - Log into the Hugging Face platform with your credentials.
+   - Click on your profile image and go to "Settings".
+   - On the left side bar menu, click the "Access Token" option.
+   - Click on "New token" to create a new token, give it a name and assign it the "Read" permissions
+   - Give your token a name, select the necessary scopes (e.g., "repo" for accessing private repositories), and click "Generate token."
+   - Store the generated token in your local machine (<a href="https://huggingface.co/docs/huggingface_hub/en/quick-start#login-command" target="_blank">official docs here</a>):
+     - Install the Hugging Face hub library: `pip install -U huggingface_hub`
+     - From the terminal run the `login()` command: `huggingface-cli login`, which will tell you if you are already logged in and prompt you for your token. The token is then validated and saved in your `HF_HOME` directory (defaults to `~/.cache/huggingface/token`).
+3. Submit your AI agent:
+   - Choose the appropriate dependencies docker image for your submission. We provide <a href="https://github.com/orgs/diambra/packages?repo_name=arena" target="_blank">different pre-built ones</a> giving access to various common third party libraries
+   - Submit your agent as shown in the following examples
+
+#### Example 1: Using a Manifest File (Recommended)
+
+Assuming you are using the `arena-stable-baselines3-on3.10-bullseye` dependencies image, create a file named `submission-manifest.yaml` with the following content:
+
+```yaml
+mode: AIvsCOM
+image: diambra/arena-stable-baselines3-on3.10-bullseye:main
+env:
+  HF_TOKEN: '{{ .Secrets.HF_TOKEN }}'
+command:
+  - python
+  - "./agent.py"
+  - "hf-repo-id"
+  - "./models/model.zip"
+```
+
+Change `image` and `command` fields according to your specific use case.
+
+Then, submit your agent using the manifest file:
+
+```sh
+diambra agent submit --submission.secrets-from=huggingface --submission.manifest submission-manifest.yaml
+```
+
+This will automatically retrieve the Hugging Face token you saved earlier.
+
+Note that your agent script must contain the instructions to download your model from Hugging Face as shown in the examples provided in our <a href="https://github.com/diambra/agents" target="_blank">DIAMBRA Agents</a> repository.
+
+#### Example 2: Command Line Interface Only
+
+If you want to avoid using submission files, you can use the command line to directly submit your agent. Assuming you are using the `arena-stable-baselines3-on3.10-bullseye` dependencies image:
+
+```sh
+diambra agent submit \
+  --submission.mode AIvsCOM \
+  --submission.secrets-from=huggingface \
+  --submission.set-command \
+  arena-stable-baselines3-on3.10-bullseye \
+  python "./agent.py" "hf-repo-id" "./models/model.zip"
```

Comment:
Do we have support for this in the agent.py here already?

Reply:
Not yet, that was a work in progress placeholder (that is also why the pr was not yet there), was waiting for your feedback on if we wanted to support the same process as git.
On this regard, do you think it make sense to have also the github-like option for HF?